src/components/rating/rating.ts

```diff
@@ -0,0 +1,297 @@
+      case 'ArrowLeft':
+        result -= this.isLTR ? this.step : -this.step;
+        break;
+      case 'Home':
+        result = this.step;
+        break;
+      case 'End':
+        result = this.max;
+        break;
+      default:
+        return;
+    }
+
+    // Verify new value is in bounds and emit
+    this.value = clamp(result, 0, this.max);
+
+    if (result === this.value) {
+      this.emitEvent('igcChange', { detail: this.value });
+    }
+  }
+
+  protected calcNewValue(x: number) {
+    const { width, left, right } = this.container.getBoundingClientRect();
+    const percent = this.isLTR ? (x - left) / width : (right - x) / width;
+    const value = this.round(this.max * percent + this.step / 2);
+
+    return clamp(value, this.step, this.max);
+  }
+
+  protected getPrecision(num: number) {
+    const [_, decimal] = num.toString().split('.');
+    return decimal ? decimal.length : 0;
+  }
+
+  protected round(value: number) {
+    value = Math.round(value / this.step) * this.step;
+    return Number(value.toFixed(this.getPrecision(this.step)));
+  }
+
+  /**
+   * Increments the value of the control by `n` steps multiplied by the
+   * step factor.
+   */
+  public stepUp(n = 1) {
+    this.value += this.round(n * this.step);
+  }
+
+  /**
+   * Decrements the value of the control by `n` steps multiplied by
+   * the step factor.
+   */
+  public stepDown(n = 1) {
+    this.value -= this.round(n * this.step);
+  }
+
+  protected renderSymbol(index: number) {
+    return this.symbolFormatter ? this.symbolFormatter(index) : this.symbol;
+  }
+
+  protected *renderSymbols() {
+    for (let i = 0; i < this.max; i++) {
+      yield html`
+        <span part="symbol ${this.size}"> ${this.renderSymbol(i)} </span>
+      `;
+    }
+  }
+
+  protected renderFractionWrapper(styles: { width: string }) {
+    return this.hoverPreview
+      ? html` <div
+          @click=${this.handleClick}
+          @mouseenter=${this.handleMouseEnter}
+          @mouseleave=${this.handleMouseLeave}
+          @mousemove=${this.handleMouseMove}
+        >
+          <div style=${styleMap(styles)} part="fraction ${this.size}">
+            <div part="symbols-wrapper fraction">${this.renderSymbols()}</div>
+          </div>
+          <div part="symbols-wrapper">${this.renderSymbols()}</div>
+        </div>`
+      : html`
+          <div @click=${this.handleClick}>
+            <div style=${styleMap(styles)} part="fraction ${this.size}">
+              <div part="symbols-wrapper fraction">${this.renderSymbols()}</div>
+            </div>
+            <div part="symbols-wrapper">${this.renderSymbols()}</div>
+          </div>
+        `;
+  }
+
+  protected override render() {
+    const value = this.hoverState ? this.hoverValue : this.value;
+    const styles = { width: `${Math.round((value / this.max) * 100)}%` };
+
+    return html`
+      <div part="rating">
+        <label part="label ${this.size}" for="">${this.label}</label>
```

Comment:
Do we need to provide built-in label? We don't provide such for the slider component.

Reply:
This comment is not addressed.

Reply:
Maybe @StefanIvanov can answer that question.

Reply:
If you look at the [rating spec](https://github.com/IgniteUI/igniteui-webcomponents/wiki/Rating-Specification) you will notice that we have in the designs a label and even a value label, but then we decided that both might be a bit too much and left the only the more agnostic one i.e. the label. I understand your concerns and reference to the slider as it practically has the opposite i.e. no such label but one or two value labels positioned on the thumbs. Now the slider design is largely affected by the default material theme where labels of form elements are subtle and to a degree unnecessary. Having added bootstrap and fluent themes they are much more prominent when it comes to labels of form elements and this is also a mechanism to establish a rhythm of the form itself. Therefore, I'd advise considering the addition of a "generic" label to the slider instead of removing it from the rating.